src/cache/mod.rs

```diff
@@ -387,39 +362,40 @@ impl Cache {
         self.guilds.iter().map(|i| *i.key()).chain(unavailable_guild_ids).collect()
     }
 
-    /// Retrieves a [`Channel`] from the cache based on the given Id.
-    ///
-    /// This will search the `channels` map, then the [`Self::private_channels`] map.
-    ///
-    /// If you know what type of channel you're looking for, you should instead manually retrieve
-    /// from one of the respective methods:
+    /// Retrieves a [`GuildChannel`] from the cache based on the given Id.
     ///
-    /// - [`GuildChannel`]: [`Self::guild_channel`] or [`Self::guild_channels`]
-    /// - [`PrivateChannel`]: [`Self::private_channel`] or [`Self::private_channels`]
+    /// This will search the guild cache for the [`Guild::channels`] map, then the
+    /// `Self::temp_channels` map if enabled and not found.
     #[inline]
-    pub fn channel<C: Into<ChannelId>>(&self, id: C) -> Option<Channel> {
+    pub fn channel<C: Into<ChannelId>>(&self, id: C) -> Option<GuildChannelRef<'_>> {
         self._channel(id.into())
     }
 
-    fn _channel(&self, id: ChannelId) -> Option<Channel> {
-        if let Some(channel) = self.guild_channel(id) {
-            return Some(Channel::Guild(channel.clone()));
+    fn _channel(&self, id: ChannelId) -> Option<GuildChannelRef<'_>> {
+        let guild_id = *self.channels.get(&id)?;
+        let guild_ref = self.guilds.get(&guild_id)?;
+        let channel = guild_ref.try_map(|g| g.channels.get(&id)).ok();
+        if let Some(channel) = channel {
+            return Some(CacheRef::from_mapped_ref(channel));
         }
 
         #[cfg(feature = "temp_cache")]
-        {
-            if let Some(channel) = self.temp_channels.get(&id) {
-                return Some(Channel::Guild(channel));
-            }
-        }
-
-        if let Some(private_channel) = self.private_channels.get(&id) {
-            return Some(Channel::Private(private_channel.clone()));
+        if let Some(channel) = self.temp_channels.get(&id) {
+            return Some(CacheRef::from_arc(channel));
```

Comment:
Minor nit, but for diff cleanliness purposes, let's keep the wrapping braces around this. Serenity is inconsistent with how it chooses to feature-gate single-expression blocks.  Sometimes it wraps them in braces and sometimes not. But the behavior is equivalent so it's something to be decided later.